src/Runner.Listener/Runner.cs

```diff
@@ -543,13 +573,25 @@ private async Task<int> RunAsync(RunnerSettings settings, bool runOnce = false)
                                     {
                                         var actionsRunServer = HostContext.CreateService<IActionsRunServer>();
                                         await actionsRunServer.ConnectAsync(new Uri(settings.ServerUrl), creds);
-                                        jobRequestMessage = await actionsRunServer.GetJobMessageAsync(messageRef.RunnerRequestId, messageQueueLoopTokenSource.Token);
+                                        jobRequestMessage =
+                                            await actionsRunServer.GetJobMessageAsync(messageRef.RunnerRequestId,
+                                                messageQueueLoopTokenSource.Token);
                                     }
                                     else
                                     {
                                         var runServer = HostContext.CreateService<IRunServer>();
                                         await runServer.ConnectAsync(new Uri(messageRef.RunServiceUrl), creds);
-                                        jobRequestMessage = await runServer.GetJobMessageAsync(messageRef.RunnerRequestId, messageQueueLoopTokenSource.Token);
+                                        try
+                                        {
+                                            jobRequestMessage =
+                                                await runServer.GetJobMessageAsync(messageRef.RunnerRequestId,
+                                                    messageQueueLoopTokenSource.Token);
+                                        }
+                                        catch (TaskOrchestrationJobAlreadyAcquiredException)
+                                        {
+                                            Trace.Info("Job is already acquired, skip this message.");
+                                            continue;
+                                        }
                                     }
```

Comment:
rest of all is formatting, this is the actual change, just added try/catch and on `TaskOrchestrationJobAlreadyAcquiredException` we `continue`